backend/overstroomik_service/main.py

```diff
@@ -17,24 +25,60 @@
 
 @app.get("/")
 async def read_root():
-    return {"Hello": "World"}
+    return Webservice(version=settings.WS_VERSION, status=settings.ERROR_GENERAL_NOER)
+
+
+@app.get('/by_rd', response_model=FloodInfo)
+async def by_rd(x: Optional[float] = None, y: Optional[float] = None) -> FloodInfo:
+
+    # initieel no error
+    status = settings.ERROR_GENERAL_NOER
+
+    # Convert rd to lat/lon
+    rijksdriehoek = Rijksdriehoek()
+    latlon = rijksdriehoek.rd_to_wgs(x, y)
+    print (len(latlon))
+    # Get address information from the PDOK services
+    pdok = PDOK()
+    status, location = await pdok.get_address(searchfield=None, latitude=latlon[0], longitude=latlon[1])
+
+    data = Data()
+    if status == settings.ERROR_GENERAL_NOER:
+        # Get location information from the geoserver
+        data = Data()
+
+    return FloodInfo(
+        webservice=Webservice(version=settings.WS_VERSION, status=status),
+        location=location,
+        data=data
+    )
 
 
 @app.get("/by_location", response_model=FloodInfo)
-def by_location(
+async def by_location(
     search_field: Optional[str] = None,
     latitude: Optional[float] = None,
     longitude: Optional[float] = None,
 ) -> FloodInfo:
-    # TODO
-    # Derive location
+
+    # initieel no error
+    status = settings.ERROR_GENERAL_NOER
```

Comment:
You always get status from get_address, so it's not needed here.

Reply:
removed from code